### DISCUSSION TITLE
[question/request for info] addon's network config info

### DISCUSSION THREAD

**Discussion by patienttruth** [hassio-addons/app-tailscale#338] [Q&A] [answered]
Greetings, this add-on works great, however I'm trying to move away from it to a seperate instance of Tailscale (TS) so that it's up regardless of HAOS. I got TS set up in Docker outside of HAOS, however I ran into issues with Chromecast which on a separate VLAN. Issues I don't have when using the TS add-on.

The issue is that I can see the Chromecast, but cast won't connect. Since I can see it, I don't think it's mDNS. Some helpful person on Reddit suggested setting up static routing, but that's not something I'm familiar with AND since it works with HAOS (on same machine (HAOS on VM and Docker running on LXC within Promox VE)) it can clearly be accomplished without getting into custom routing tables.

More background here in the Reddit post - https://www.reddit.com/r/homeassistant/comments/1anbeu9/home_assistant_vs_docker_chromecast_connection/

My question is,  is there anything related to network routing that is implemented in this add-on that isn't implemented by default in a separate docker instance of TS?

I know this is outside the scope of an issue with the add-on, so any help is greatly appreciated. And if there's a better place to ask this question, please let me know.

**Comment by lmagyar** [comment]
> My question is, is there anything related to network routing that is implemented in this add-on that isn't implemented by default in a separate docker instance of TS?

It depends on the add-on config. Could you please copy the yaml here (without any possibly sensitive content)?

**Reply by patienttruth** [reply]
```accept_dns: true
advertise_exit_node: true
advertise_routes:
  - 192.168.LAN.0/23
  - 192.168.IOT.0/23
log_level: debug
```

As far as I can tell I set up the docker instance with the same settings.

**Reply by lmagyar** [reply]
That's nothing special. When userspace networking is enabled, the add-on doesn't modify routing/networking settings.

Please read the add-on docs, and check, that all the default enabled options (not only advertise-exit-node) are also enabled in your container:
  - tailscaled --tun=userspace-networking (more details: https://github.com/hassio-addons/addon-tailscale/blob/main/tailscale/rootfs/etc/s6-overlay/s6-rc.d/tailscaled/run)
  - tailscale up --accept-dns --accept-routes --snat-subnet-routes  (more details: https://github.com/hassio-addons/addon-tailscale/blob/main/tailscale/rootfs/etc/s6-overlay/s6-rc.d/post-tailscaled/run)

There are other internal services in the add-on (doing some routing black magic), but they won't start with your config.

As a last resort, at HA cli (SSH add-on): ``docker exec -it `docker ps -q -f name=tailscale` /bin/bash`` then inside the add-on `/opt/tailscale status --json --peers=false` and compare it with your tailscale container status, maybe you can spot some relevant difference.

**Reply by patienttruth** [reply, marked as answer]
@lmagyar first off, thanks for the reply. It prompted diving into comparison of the inspection reports between the HAOS addon and my docker instance. (might I add that I have a heck of a time copying the text out of the HA cli, haven't figured that out).

You know what did it? Specifying a DNS server to the TS container. I noticed that there wasn't one specified, and at the time I believe the Docker LXC was set to inherit DNS from the Proxmox node which had 8.8.8.8. I added my internal DNS by adding:
```
    dns:
      - 192.168.X.X
```
to my docker-compose and it started working.

You know I've read "it's always DNS", but it's not always obvious... :)
Thanks!
